sdks/go/pkg/beam/core/runtime/exec/coder.go

```diff
@@ -276,42 +277,135 @@ func (c *kvDecoder) Decode(r io.Reader) (FullValue, error) {
 
 }
 
-// TODO(herohde) 4/7/2017: actually handle windows.
+// WindowEncoder handles Window serialization to a byte stream. The encoder
+// can be reused, even if an error is encountered.
```

Comment:
Any concurrency-safety requirements?

Reply:
Added comment to all coding types that they are concurrency-safe.